internal/feature_test.go

```diff
@@ -110,8 +111,17 @@ func TestScriptBasedInit(t *testing.T) {
 
 }
 
-// TestScriptBasedCore tests ScriptBasedSoftwareUpdatable core functionality: init, install and download.
-func TestScriptBasedCore(t *testing.T) {
+// TestScriptBasedDownloadAndInstall tests ScriptBasedSoftwareUpdatable core functionality: init, install and download operations.
+func TestScriptBasedDownloadAndInstall(t *testing.T) {
+	testScriptBasedSoftwareUpdatableOperations(true, t)
+}
+
+// TestScriptBasedDownloadAndInstallResume tests ScriptBasedSoftwareUpdatable core functionality: init, resuming install and download operations.
+func TestScriptBasedDownloadAndInstallResume(t *testing.T) {
+	testScriptBasedSoftwareUpdatableOperations(false, t)
+}
+
+func testScriptBasedSoftwareUpdatableOperations(basic bool, t *testing.T) {
```

Comment:
basic - the parameter name is not very informative, it should be noResume or resume but the logic has to be fixed